apps/mobile/src/modules/documents-actions/components/document-action-sheet.tsx

```diff
@@ -0,0 +1,319 @@
+import type { ThemeColors } from '@/modules/ui/theme.constants';
+import {
+  Modal,
+  View,
+  Text,
+  StyleSheet,
+  TouchableOpacity,
+  TouchableWithoutFeedback
+} from 'react-native';
+import { type CoerceDate } from '@/modules/api/api.models';
+import { MaterialCommunityIcons } from '@expo/vector-icons';
+import { useThemeColor } from '@/modules/ui/providers/use-theme-color';
+import type { Document } from '@/modules/documents/documents.types';
+import { useApiClient, useAuthClient } from '@/modules/api/providers/api.provider';
+import { fetchDocumentFile } from '@/modules/documents/documents.services';
+import * as Sharing from 'expo-sharing';
+import { useAlert } from '@/modules/ui/providers/alert-provider';
+import { configLocalStorage } from '@/modules/config/config.local-storage';
+
+interface DocumentActionSheetProps {
+  visible: boolean;
+  document: CoerceDate<Document> | undefined;
+  onClose: () => void;
+  onView: () => void;
+  onDownloadAndShare: () => void;
+}
+
+export function DocumentActionSheet({
+  visible,
+  document,
+  onClose,
+  onView,
+  onDownloadAndShare,
+}: DocumentActionSheetProps) {
+  const themeColors = useThemeColor();
+  const styles = createStyles({ themeColors });
+  const { showAlert } = useAlert();
+  const authClient = useAuthClient();
+
+  if (document == undefined) return null;
+
+  // Check if document can be viewed in DocumentViewerScreen
+  // Supported types: images (image/*) and PDFs (application/pdf)
+  const isViewable =
+    document.mimeType.startsWith('image/') ||
+    document.mimeType.startsWith('application/pdf');
+
+  const formatFileSize = (bytes: number): string => {
+    if (bytes < 1024) return `${bytes} B`;
+    if (bytes < 1024 * 1024) return `${(bytes / 1024).toFixed(1)} KB`;
+    return `${(bytes / (1024 * 1024)).toFixed(1)} MB`;
+  };
+
+  const formatDate = (dateString: string): string => {
+    const date = new Date(dateString);
+    return date.toLocaleDateString('en-US', {
+      month: 'short',
+      day: 'numeric',
+      year: 'numeric',
+    });
+  };
+
+  const handleDownloadAndShare = async () => {
+    const baseUrl = await configLocalStorage.getApiServerBaseUrl();
+
+    if (!baseUrl) {
+      showAlert({
+        title: 'Error',
+        message: 'Base URL not found',
+      });
+      return;
+    }
+
+    try {
+      const fileUri = await fetchDocumentFile({
+        document: document,
+        organizationId: document.organizationId,
+        baseUrl,
+        authClient,
+      });
+
+      const canShare = await Sharing.isAvailableAsync();
+      if (canShare) {
+        await Sharing.shareAsync(fileUri);
+      } else {
+        showAlert({
+          title: 'Sharing Failed',
+          message: 'Sharing is not available on this device. Please share the document manually.',
+        });
+      }
+
+    } catch (error) {
```

Comment:
Check for the availability of the sharing api before fetching the file to avoid unnecessary transfer